File: flutter_lockbox/android/src/main/kotlin/com/lockbox/sdk/flutter_lockbox/FlutterLockboxPlugin.kt
package com.lockbox.sdk.flutter_lockbox

import android.content.Context
import com.lockbox.sdk.core.Lockbox
import io.flutter.embedding.engine.plugins.FlutterPlugin

/**
 * FlutterLockboxPlugin
 * Bridges Flutter Pigeon calls to the native Android Core Lockbox library.
 */
class FlutterLockboxPlugin : FlutterPlugin, LockboxHostApi {

    private var context: Context? = null

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        context = binding.applicationContext
        LockboxHostApi.setUp(binding.binaryMessenger, this)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        context = null
        LockboxHostApi.setUp(binding.binaryMessenger, null)
    }

    // --- LockboxHostApi Implementation ---

    override fun saveSecureString(key: String, value: String): Boolean {
        val currentContext = context ?: return false
        return try {
            Lockbox.instance.saveSecureString(currentContext, key, value)
            true
        } catch (e: Exception) {
            false
        }
    }

    override fun getSecureString(key: String): String? {
        val currentContext = context ?: return null
        return try {
            Lockbox.instance.getSecureString(currentContext, key)
        } catch (e: Exception) {
            null
        }
    }

    override fun deleteSecureString(key: String): Boolean {
        val currentContext = context ?: return false
        return try {
            Lockbox.instance.deleteSecureString(currentContext, key)
            true
        } catch (e: Exception) {
            false
        }
    }
}

File: flutter_lockbox/ios/Classes/FlutterLockboxPlugin.swift
import Flutter
import UIKit
import Lockbox

/**
 * FlutterLockboxPlugin
 * Bridges Flutter Pigeon calls to the native iOS Core Lockbox library.
 */
public class FlutterLockboxPlugin: NSObject, FlutterPlugin, LockboxHostApi {

  public static func register(with registrar: FlutterPluginRegistrar) {
    let instance = FlutterLockboxPlugin()
    LockboxHostApiSetup.setUp(binaryMessenger: registrar.messenger(), api: instance)
  }

  // --- LockboxHostApi protocol implementation ---

  public func saveSecureString(key: String, value: String) throws -> Bool {
    return Lockbox.shared.saveSecureString(key, value: value)
  }

  public func getSecureString(key: String) throws -> String? {
    return Lockbox.shared.getSecureString(key)
  }

  public func deleteSecureString(key: String) throws -> Bool {
    return Lockbox.shared.deleteSecureString(key)
  }
}
